app/src/androidTest/java/com/sdpteam/connectout/GreetingsActivityTest.java

```diff
@@ -25,12 +29,43 @@ public class GreetingsActivity {
     @Test
     public void testGreetingActivity() {
         Intent intent = new Intent(ApplicationProvider.getApplicationContext(), GreetingActivity.class);
-        intent.putExtra("key", "coucou lol");
+        intent.putExtra("loginInfo", "coucou lol");
         activityScenarioRule.getScenario().onActivity(activity -> {
             activity.startActivity(intent);
         });
 
         onView(withId(R.id.greetingMessage)).check(matches(isDisplayed()));
         onView(withId(R.id.greetingMessage)).check(matches(ViewMatchers.withText("coucou lol")));
```

Comment:
Looks good ! However,  separating the checks into two different tests would make it what is tested more clear.
I would advise changing the test name too